### PR TITLE
Add support for eclipse variable length strings

### PR DESCRIPTION
**Pull request by markusdregi**
Implement support for variable length eclipse strings

### CODE REVIEW

**Comment by joakim-hove** [comment]
Please remove all occurences of the `C010` datatype.

**Comment by joakim-hove** [comment]
Testing:

1. Just create a temporary area and write a couple of keywords and read them back again.
2. This test: https://github.com/Statoil/libecl/blob/master/python/tests/core/ecl/test_sum_statoil.py#L454 should be updated to verify that an actual summary case is now loaded correctly.

File: libecl/src/ecl_kw.c [hunk 1]
```diff
@@ -161,61 +161,62 @@ UTIL_IS_INSTANCE_FUNCTION(ecl_kw , ECL_KW_TYPE_ID )
 ecl_type_enum  ecl_kw_get_type(const ecl_kw_type *);
 void ecl_kw_set_data_type(ecl_kw_type * ecl_kw, ecl_data_type data_type);
 
-static const char * get_read_fmt(const ecl_data_type data_type ) {
+static char * alloc_read_fmt_string(const ecl_data_type ecl_type) {
+  char fmt [6];
+  sprintf(fmt, "%%%dc", ecl_type_get_sizeof_ctype_fortio(ecl_type));
```
Comment (joakim-hove) [inline libecl/src/ecl_kw.c]: Here you can use:
```C
return util_alloc_sprint("%%%dc", ecl_type_get_sizeof_ctype_fortio(ecl_type));
```

File: python/tests/ecl/test_ecl_kw.py [hunk 2]
```diff
@@ -416,3 +419,39 @@ def test_string_alloc(self):
...
+    def test_string_write_read_unformatted(self):
+        for str_len in range(1000):
+            with TestAreaContext("my_space"):
+
+                kw = EclKW("TEST_KW" , 10, EclDataType.ECL_STRING(str_len))
+                for i in range(10):
+                    kw[i] = str(i)*str_len
+
+                file_name = "ecl_kw_test"
+                fortio = FortIO(file_name, mode=FortIO.WRITE_MODE)
```
Comment (joakim-hove) [inline python/tests/ecl/test_ecl_kw.py]: Here you can use the `FortIO` context:

```python
with openFortIO( file_name , mode = FortIO.WRITE_MODE) as f:
    kw.fwrite( f )
```

File: libecl/src/ecl_kw.c (hunk 1 above)
Comment (markusdregi) [inline libecl/src/ecl_kw.c]: Cool! That is useful for me several places..

**Comment by joakim-hove** [comment]
As in *not* WIP any longer?

**Comment by markusdregi** [comment]
@joakim-hove I was planning to leave it as WIP until the tests passed. But yes, I can remove the WIP now..

**Pull request merged**